daemon/graphdriver/lcow/lcow.go

```diff
@@ -102,10 +117,36 @@ func (d *Driver) startUvm(context string) error {
 		return fmt.Errorf("failed to generate default gogcs configuration (%s): %s", context, err)
 	}
 
+	scratchAttached := false
+	if _, err := os.Stat(d.svmScratchSpaceFile); err == nil {
+		// We have a scratch space already, so just attach it as a mapped virtual disk
+		logrus.Debugf("lcowdriver: startuvm: (%s) attaching pre-existing scratch", context)
+		mvd := hcsshim.MappedVirtualDisk{
+			HostPath:          d.svmScratchSpaceFile,
+			ContainerPath:     `/mnt/gcs/LinuxServiceVM/scratch`,
+			CreateInUtilityVM: true,
+		}
+		d.config.MappedVirtualDisks = append(d.config.MappedVirtualDisks, mvd)
+		scratchAttached = true
+	}
+
 	d.config.Name = "LinuxServiceVM" // TODO @jhowardmsft - This requires an in-flight platform change. Can't hard code it to this longer term
 	if err := d.config.Create(); err != nil {
 		return fmt.Errorf("failed to start utility VM (%s): %s", context, err)
 	}
+
+	// Hot-add the scratch-space if not already attached
+	if !scratchAttached {
+		logrus.Debugf("lcowdriver: startuvm: (%s) creating an SVM scratch", context)
+		if err := d.config.CreateSandbox(d.svmScratchSpaceFile, client.DefaultSandboxSizeMB, d.cachedSandboxFile); err != nil {
+			return fmt.Errorf("failed to create SVM scratch VHDX (%s): %s", context, err)
+		}
+		logrus.Debugf("lcowdriver: startuvm: (%s) hot-adding an SVM scratch", context)
+		if err := d.config.HotAddVhd(d.svmScratchSpaceFile, `/mnt/gcs/LinuxServiceVM/scratch`); err != nil {
```

Comment:
Agreed. However this code will change shortly to have a single service VM. This is just a single step forward.